kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/db/H2DbServiceImpl.java

```diff
@@ -154,20 +162,36 @@ public Connection getConnection() throws SQLException {
             syncWithExecutor();
         }
 
-        final Lock lock = this.rwLock.readLock();
-        lock.lock();
-        Connection connection = null;
+        Future<T> result = this.executorService.submit(() -> {
```

Comment:
Should we submit the task with the `readLock` held? Can this solve the issues in the comments above?